File: csharp/861.score-after-flipping-matrix.cs
/*
 * @lc app=leetcode id=861 lang=csharp
 *
 * [861] Score After Flipping Matrix
 */

using System;

// @lc code=start
public class Solution {
    public int MatrixScore(int[][] A) {
        int m = A.Length, n = A[0].Length, ret = m * (1 << (n - 1));
        for (int i = 1; i < n; ++i) {
            int flips = 0;
            for (int j = 0; j < m; ++j) {
                if (A[j][0] != A[j][i]) {
                    ++flips;
                }
            }
            ret += Math.Max(flips, m - flips) * (1 << (n - 1 - i));
        }
        return ret;
    }
}
// @lc code=end

File: java/861.score-after-flipping-matrix.java
/*
 * @lc app=leetcode id=861 lang=java
 *
 * [861] Score After Flipping Matrix
 */

// @lc code=start
class Solution {
    public int matrixScore(int[][] A) {
        int m = A.length, n = A[0].length, ret = (1 << (n - 1)) * m;
        for (int j = 1; j < n; ++j) {
            int diff = 0;
            for (int i = 0; i < m; ++i) {
                diff += A[i][0] ^ A[i][j];
            }
            ret += (1 << (n - 1 - j)) * Math.max(diff, m - diff);
        }
        return ret;
    }
}
// @lc code=end

